packages/ubuntu_bootstrap/lib/pages/source/codecs_and_drivers_page.dart

```diff
@@ -13,75 +14,255 @@ export 'source_model.dart' show kFullSourceId, kMinimalSourceId;
+  // Focus nodes for checkboxes
+  final FocusNode _driversFocusNode = FocusNode();
+  final FocusNode _codecsFocusNode = FocusNode();
+
+  @override
+  void initState() {
+    super.initState();
+    
+    // Announce page on load
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (!mounted) return;
+      
+      final lang = UbuntuBootstrapLocalizations.of(context);
+      final model = ref.read(sourceModelProvider);
+      
+      // Use PageAnnouncer for consistent semantics handling
+      PageAnnouncer.announcePageLoad(
+        title: lang.codecsAndDriversPageTitle,
+        subtitle: lang.codecsAndDriversPageDescription,
+        instructions: lang.codecsAndDriversPageInstructions,
+      );
+      
+      // Set focus on first checkbox
+      Future.delayed(const Duration(milliseconds: 500), () {
+        if (mounted) {
+          _driversFocusNode.requestFocus();
+        }
+      });
+      
+      // Announce if on battery
+      if (model.onBattery) {
+        Future.delayed(const Duration(milliseconds: 800), () {
+          if (mounted) {
+            SemanticsService.announce(
+              lang.sourceBatteryWarning,
+              TextDirection.ltr,
+            );
+          }
+        });
+      }
+      
+      // Announce if offline and codecs are disabled
+      if (!model.isOnline) {
+        Future.delayed(const Duration(milliseconds: 1000), () {
+          if (mounted) {
+            SemanticsService.announce(
+              lang.offlineWarning,
+              TextDirection.ltr,
+            );
+          }
+        });
+      }
+    });
+  }
+
+  @override
+  void dispose() {
+    _driversFocusNode.dispose();
+    _codecsFocusNode.dispose();
+    super.dispose();
+  }
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
+  Widget build(BuildContext context) {
     final model = ref.watch(sourceModelProvider);
     final lang = UbuntuBootstrapLocalizations.of(context);
     final flavor = ref.watch(flavorProvider);
 
-    return HorizontalPage(
-      windowTitle: lang.codecsAndDriversPageTitle,
-      title: lang.codecsAndDriversPageDescription,
-      snackBar: model.onBattery ? OnBatterySnackBar() : null,
-      isNextEnabled: model.sourceId != null,
-      onNext: () async {
-        final telemetry = tryGetService<TelemetryService>();
-        await telemetry?.addMetrics({
-          'RestrictedAddons': model.installCodecs,
-        });
-        await model.save();
-      },
-      children: [
-        Text(lang.codecsAndDriversPageBody(flavor.displayName)),
-        const SizedBox(height: kWizardSpacing),
-        // TODO(Lukas): Add a proper check when we know where to get this information.
-        if (false)
-          // ignore: dead_code
-          InfoBox(
-            title: lang.codecsAndDriversNvidiaNote,
-            subtitle: lang.codecsAndDriversNvidiaBody,
-            isThreeLine: true,
+    return Semantics(
+      label: 'Proprietary software installation page',
```

Comment:
There still are areas where there are normal strings instead of localized strings. Please make sure that every instance of human-readable strings (including semantic labels and announcements) are localized.